File: app/src/main/java/com/example/testingapp/MainActivity.java
package com.example.testingapp;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProvider;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.example.testingapp.databinding.ActivityMainBinding;
import com.example.testingapp.roomdb.User;
import com.example.testingapp.roomdb.UserDao;
import com.example.testingapp.roomdb.UserDatabase;
import com.example.testingapp.sqlite.SqlMainActivity;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    private Datac datac;
    private UserDao mUserDao;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        /**
         * 뷰 바인드 , MVVM 실습
         */
        //setContentView(R.layout.activity_main);

       /* datac= new Datac();
        ActivityMainBinding activityMainBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);

        activityMainBinding.setActVar(datac);
        activityMainBinding.setLifecycleOwner(this);*/

      /*  activityMainBinding.btnChange.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

            }
        });
       */

        /**
         * Room DB 실습
         * 1.데이터 entity 모델 , Dao 인터페이스, RoomDB 추상클래스 생성
         * 2.RoomDB 추상클래스에 dao, entity 매핑
         * 3.db 빌더를 통해 db객체 빌드
         * 4.dao 인터페이스에 db객체 할당
         * 5.CRUD Operation
         */


      /*  UserDatabase userDatabase = Room.databaseBuilder(getApplicationContext(), UserDatabase.class , "User_DB")
                .fallbackToDestructiveMigration() //스키마 (Database)버전 변경 가능
                .allowMainThreadQueries()   //Main Thread에서 DB IO(입출력) 가능하게 함
                .build();



        mUserDao = userDatabase.userDao(); //인터페이스 객체 할당

        //데이터 삽입
        User user = new User("샘플1","22","2222-4444"); //
        mUserDao.setInsertUser(user);

        //데이터 조회
        List<User> userList = mUserDao.getUserAll();
        for(int i = 0 ; i<userList.size() ; i++){
            Log.d("TEST", userList.get(i).getName());
        }
*/

        /**
         * SqLite DB Connect
         */


    }
}

File: app/src/main/java/com/mjh/lovetest/MainActivity.kt
package com.mjh.lovetest

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.FragmentContainer
import androidx.fragment.app.FragmentContainerView
import androidx.navigation.NavController
import androidx.navigation.findNavController
import androidx.navigation.fragment.NavHostFragment
import androidx.navigation.fragment.findNavController
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(){

    lateinit var navController: NavController
    lateinit var navHostFragment :NavHostFragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        //네비게이션 바인딩
//        var fragmentContainerView : FragmentContainerView  = findViewById(R.id.nav_host_fragment)
//        navController = nav_host_fragment.findNavController()

        /**
         * 기본적으로 "as" 연산자는 대상 값을 as로 지정한 타입으로 캐스트 하며, 해당 타입으로 바꿀 수 없으면 ClassCastException이 발생하게 됩니다.
        이럴 경우 is 연산자를 통해 대상 값이 해당 타입으로 변환 가능한 타입인지 체크해야 하지만 코틀린에서는 훨씬 더 간결한 기능으로 "as?" 연산자를 제공합니다.
        "as?" 연산자는 어떤 값을 지정한 타입으로 캐스트를 하고, 만약 대상 타입으로 캐스트 할 수 없으면 null을 반환하게 됩니다.
         */

        /**
         * 우리는 FragmentContainerView를 사용하여 NavHostFragment를 만들었다.
        위와 같이 supportFragmentManager에서 findFragmentById를 사용해 직접 해당 FragmentContainerView를 검색해서
        해당 View가 지닌 Class속성인 navHostFragment로 TypeCasting을 해야 가져올 수 있다.
         */
        navHostFragment = supportFragmentManager.findFragmentById(R.id.nav_host_fragment) as NavHostFragment
//        var navController = navHostFragment.navController
//        navController = nav_host_fragment.findNavController()
        navController = navHostFragment.findNavController()



    }
}